language/transaction-builder/src/lib.rs

```diff
@@ -121,12 +121,10 @@ pub fn encode_rotate_consensus_pubkey_script(new_key: Vec<u8>) -> Script {
 }
 
 /// Encode a program that rotates the sender's authentication key to `new_key`.
-pub fn rotate_authentication_key_script(new_key: AccountAddress) -> Script {
+pub fn rotate_authentication_key_script(new_key: Vec<u8>) -> Script {
```

Comment:
nit: thoughts on renaming this to `new_key_hash` and specifying the kind of hashed bytes we expect here?  Just to make it explicit that we should not pass a raw public key.